src/main/scala/org/apache/spark/sql/delta/catalog/DeltaTableV2.scala

```diff
@@ -47,11 +47,16 @@ case class DeltaTableV2(
     specifiedSchema: Option[StructType] = None,
     specifiedPartitioning: Array[Transform] = Array.empty,
     tableIdentifier: Option[String] = None) extends Table with SupportsWrite {
+
+  private def metadata = log.update().metadata
```

Comment:
Calling update() every time we want to read metadata is gonna be pretty expensive.

Reply:
probably should be a lazy val then